src/prototype_room_my.js

```diff
@@ -200,6 +200,49 @@ Room.prototype.handleScout = function() {
   }
 };
 
+Room.prototype.checkForEnergyTransfer = function() {
+  let carryHelpInterval = config.carryHelpers.ticksUntilHelpCheck;
+  if (Game.time % carryHelpInterval) {
+
+    let factor = config.carryHelpers.factor;
+    this.memory.energyAvailable = (1 - factor) * this.memory.energyAvailable + (factor) * this.energyAvailable;
+    this.memory.energyAvailableSum += this.memory.energyAvailable;
+    return;
+  }
+
+  Memory.needEnergyRooms = Memory.needEnergyRooms || [];
+  this.memory.energyAvailableSum = this.memory.energyAvailableSum || 0;
+  let needHelp = this.memory.needHelp;
+  if (!needHelp && this.memory.energyAvailableSum < config.carryHelpers.needTreshold * carryHelpInterval && !this.hostile && Game.rooms[this.name].find(STRUCTURE_SPAWN)) {
+    this.memory.energyAvailableSum = 0;
+    Memory.needEnergyRooms.push(this.name);
+    this.memory.needHelp = true;
+    return '---!!!---' + this.name + ' need energy ---!!!---';
+  } else if (needHelp) {
+    _.remove(Memory.needEnergyRooms, (r) => r === this.name);
+    delete Memory.rooms[this.name].needHelp;
```

Comment:
yeah because the if check first if it need help. so if it come here, it don't need help